### DISCUSSION TITLE
How to ensure to increase ID monotonically in ORACLE DB with RAC?

### DISCUSSION THREAD

**Discussion by indiVar0508** [sqlalchemy/sqlalchemy#10195] [Usage Questions] [answered]
Hi

I had a question regarding, sequence behavior specifically for Oracle DB, in the package that i am using i have a table that tracks the ID
for each flush i.e. whenever a flush happens it will make a entry to this Table ( this table has a col called id which is PK and is autoincrementing),

Now oracle Db with RAC ( ref: https://docs.oracle.com/cd/B14117_01/server.101/b10759/statements_6014.htm ), oracle assign sequence of Pk(s) to individual client so say like if client 1 and client 2 are working at same time it would be like oracle would assign PK range of 1-50 to client 1 and 500-550 to client 2.

issue occurs when client 1 and client 2 do alternate commits 
	client 1 commits gives PK 1 (at t0)
	client 2 commits gives PK 500 (at t1)
	client 1 commits gives PK 2 (at t2)
	client 2 commits gives PK 501 (at t3)

i wanted to have these id increase monotonically always according to timestamp, i was wondering if there is some sa.Sequence setting or extened class
that ensures this monotonical increasing of ID(s) based order of commit/flush or some hook that i could listen to that will help me create a id and assign for concurrent client

**Comment by indiVar0508** [comment, marked as answer]
> 

It seems there is parameter specifying same 
 https://github.com/sqlalchemy/sqlalchemy/blob/1377f33075a0e4230c88506d2bfd1e847a336da5/lib/sqlalchemy/sql/schema.py#L3707 parameter available to specify same.

**Reply by CaselIT** [reply]
for new table it may also worth looking into generated as identity, assuming it's available on your db version https://docs.oracle.com/en/database/other-databases/nosql-database/21.2/sqlreferencefornosql/creating-tables-identity-column.html#GUID-8E4A6DBF-1677-406C-B3A5-FD523A0C2B56
